pingcap/tidb-operator
# Set labels for TiProxy instances

### What problem does this PR solve?


TiProxy needs to route to the local TiDB instances in the same AZ. To implement that, TiProxy needs to know its own AZ.

### What is changed and how does it work?


- Set the labels to TiProxy by tiproxyctl when the TiProxy version >= v1.1.0. Most of the code is copied from TiDBMemberManager.
- Fix the URL that tiproxyctl uses. It used the server address, not pod addresses. Check `defaultTiProxyControl`.

### Code changes


### Tests



Manual test steps:
1. Start a tidb cluster with tiproxy
2. Set the location-labels on PD: `./pd-ctl config set replication.location-labels "zone"`
3. Label the node: `kubectl label node minikube zone="east"`
4. Check the label of TiProxy:
```bash
$ curl 127.1:3080/api/admin/config/ | grep zone
zone = 'east'
```

### Side effects


### Related changes


### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note
- Support automatically set location labels for TiProxy
```
